src/metric_segment_wrapper.ts

```diff
@@ -19,18 +20,32 @@ export function metricSegmentWrapper($compile, $sce, templateSrv) {
       },
 
     link: ($scope, elem) => {
-      // const $baseElem = $(metricSegmentTemplate);
       const $baseElem = $(metricSegmentTemplate);
       $baseElem.appendTo(elem);
       $compile(elem.contents())($scope);
-      if ($scope.segment.focus === true) {
-        $baseElem[0].childNodes.forEach(child => {
-          if (child.tagName === "A") {
-            setTimeout(function() { child.click(); }, 50);
-          }
-        });
-      }
+      const input = $($baseElem[0].childNodes[0]);
+
+      const typeahead = input.data("typeahead");
+      const oldSource = typeahead.source;
+
+      $scope.source = (query, callback) => {
```

Comment:
this is wrapping functionality of the inner metric-segment element so that the rendering of a remove button is not blocked by the query to get tag-values